R/helpers.R

```diff
@@ -215,3 +215,54 @@ create_dummy_parent_package <- function(mn, children) {
   list(parent = create_response,
        children = children)
 }
+
+
+#' Create dummy attributes data frame
+#'
+#' @param numberAttributes (integer) Number of attributes to be created in the table
+#' @param factors (character) Optional vector of factor names to include.
+#'
+#' @return dataframe
+#' @export
+#'
+#' @examples
+create_dummy_attributes_dataframe <- function(numberAttributes, factors=NULL) {
+  names <- sapply(1:numberAttributes, function(x){ paste0("Attribute ", x) })
+  
+  if(!is.null(factors)) {
+    domains <- c(rep("textDomain", numberAttributes - length(factors)),
+                 rep("enumeratedDomain", length(factors)))
+    names[(numberAttributes - length(factors) + 1):numberAttributes] <- factors
+  }
+  
+  df <- data.frame(attributeName = names,
+                   attributeDefinition = names,
+                   measurementScale = rep("nominal", numberAttributes),
+                   domain = rep("textDomain", numberAttributes),
+                   formatString = rep("NA", numberAttributes),
+                   definition = names,
+                   unit = rep("NA", numberAttributes),
+                   numberType = rep("NA", numberAttributes),
+                   missingValueCode = rep("NA", numberAttributes),
+                   missingValueCodeExplanation = rep("NA", numberAttributes),
+                   stringsAsFactors = F)
```

Comment:
I don't like abbreviating `TRUE`/`FALSE` to `T`/`F`. R lets you be real lax about argument names and values like `TRUE ` and `FALSE`. The reason to avoid abbreviating include:

- Confusing users with multiple ways of doing the same thing
- Relying on implicit behavior that is more likely to change in the future then the meaning of `TRUE` and `FALSE` in the future.